IBX-11536: MCP Servers (#3106)
ezsystems/developer-documentation: docs/ai/mcp/mcp_guide.md

---
description: TODO.
month_change: true
---

# Model Context Protocol and [[= product_name_base =]] MCP Servers

> [Model Context Protocol (MCP)](https://en.wikipedia.org/wiki/Model_Context_Protocol) is a protocol standardizing interactions between AIs and systems.

Author: MCP definition ressources:
- https://en.wikipedia.org/wiki/Model_Context_Protocol
- https://modelcontextprotocol.io/docs/getting-started/intro < official
- https://docs.github.com/en/copilot/concepts/context/mcp
- https://code.claude.com/docs/en/mcp
- https://developers.openai.com/api/docs/mcp (doesn't define MCP)

To target the official doc is probably better, that's what other ressources do:

```suggestion
[Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a protocol standardizing interactions between AIs and systems.
```

---

> description: MCP servers expose functionalities to AIs.

Reviewer: ```suggestion
description: MCP servers expose data sources, tools, and workflows to AI agents.
```
Or AI applications?

From https://modelcontextprotocol.io/docs/getting-started/intro
```
Using MCP, AI applications like Claude or ChatGPT can connect to data sources (e.g. local files, databases), tools (e.g. search engines, calculators) and workflows (e.g. specialized prompts)—enabling them to access key information and perform tasks.
```

Author: ```suggestion
description: MCP servers expose tools, specialized prompts, and ressources to AI agents.
```

month_change: true
---

> # Model Context Protocol and Ibexa MCP servers

Reviewer: Is this an LTS update? I think it should be - should be part of the LTS updates table?

Please add the mcp package to the list in PHP API reference script.

What is the availability? headless, exp, commerce, starting with DXP v5.0.8?

There's a template we usually follow with the guides, addressing these questions.

Author: Moved to LTS Update in cacad72ff999f289b88a9623024793002f668131

[Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a protocol standardizing interactions between AIs and systems.

While [AI actions](ai_actions_guide.md) integrate AI to the back office,

> [[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer a web interface usable by AIs outside the system.

Reviewer: ```suggestion
[[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer an API usable by AIs outside the system.
```

> [[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer a web interface usable by AIs outside the system.

Reviewer: ```suggestion
[[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer a web interface usable by AI agents outside the system.
```
or AI applications

Author: ```suggestion
[[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer an API usable by AI agents outside the system.
```

Some AI agents can use directly REST API or GraphQL API if their users explain to them how to do it in prompts, in skill files, etc.
MCP servers ease the discovery of the functionalities by AIs and help them to interpret natural language prompts into actions on the system.

`ibexa/mcp` package provides:

- MCP servers [creation by configuration](mcp_config.md#mcp-server-configuration)
- [buit-in tools](mcp_config.md#built-in-tools) to associate to MCP servers by configuration
- a PHP API to [create custom MCP server capabilities](mcp_config.md#create-capability-class)

MCP servers capabilities (tools, prompts, and resources) can be created and associated to MCP servers thanks to a PHP API mainly based on attributes.

> MCP servers are configured per repository then enabled per SiteAccess scope, allowing for flexible configurations adapted to different contexts.

Reviewer: I'd focus not on what the feature provides, but what can someone do with them - somehting like this:

```suggestion
With the MCP servers feature, you can:

- create MCP servers [by using YAML configuration](mcp_config.md#mcp-server-configuration)
- assign different [tools](mcp_config.md#built-in-tools), prompts, and resources to different MCP servers, varying them for each site
- [create custom server capabilities](mcp_config.md#create-capability-class) with PHP API

MCP servers are configured per repository and assigned to Site Accesses, allowing for flexible configurations adapted to different contexts.
```

Also, links to "repository" and "Site Access"?

Author: ```suggestion
With the MCP servers feature, you can:

- create MCP servers [by using YAML configuration](mcp_config.md#mcp-server-configuration)
- assign different [tools](mcp_config.md#built-in-tools), prompts, and resources to different MCP servers, varying them for each site
- [create custom server capabilities](mcp_config.md#create-capability-class) with PHP API

MCP servers are configured per [repository](repository_configuration.md) and assigned to [SiteAccesses](siteaccess.md), allowing for flexible configurations adapted to different contexts.
```

---
description: MCP servers expose tools, specialized prompts, and ressources to AI agents.
edition: lts-update
month_change: true
---

# Model Context Protocol and Ibexa MCP servers

[Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a protocol standardizing interactions between AIs and systems.

While [AI actions](ai_actions_guide.md) integrate AI to the back office,
[[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer an API usable by AI agents outside the system.

> Some AI agents can use directly REST API or GraphQL API if their users explain to them how to do it in prompts, in skill files, etc.

Reviewer 2: Maybe it's worth mentioning the advantages of MCP vs REST - here is unified cheatsheet: https://glama.ai/blog/2025-06-06-mcp-vs-api

Author: Thanks for this resource. I should be able to strengthen the next sentence.

Author: I reused and insisted on few ideas in f6d12d84ce4aec615c06c0233f5b9b8cbf3671c6

MCP servers ease the discovery of the functionalities by AIs and help them to interpret natural language prompts into actions on the system.

The MCP servers feature is an [LTS Update package](editions.md#lts-updates) available since v5.0.8 to all editions.

With the MCP servers feature, you can:

- create MCP servers [by using YAML configuration](mcp_config.md#mcp-server-configuration)

> - assign different [tools](mcp_config.md#built-in-tools), prompts, and resources to different MCP servers, varying them for each site

Author: ```suggestion
- assign different tools, prompts, and resources to different MCP servers, varying them for each site
- assign [built-in tools](mcp_config.md#built-in-tools) included in the package
```

- [create custom server capabilities](mcp_usage.md#create-capability-class) with PHP API

> 

Reviewer 2: I would add the information that 'you can utilize predefined Ibexa tools from src/lib/Tools directory by adding them to your server' - obviously with better wording.

Author: I can split to emphase on the built-in tools link.
See https://github.com/ibexa/documentation-developer/pull/3106/changes#r3234431059

---

> description: MCP servers expose tools, specialized prompts, and ressources to AI agents.

Reviewer 3: ```suggestion
description: MCP servers expose tools, specialized prompts, and resources to AI agents.
```

edition: lts-update
month_change: true
---

> # Model Context Protocol and Ibexa MCP servers

Reviewer 3: ```suggestion
# MCP Servers product guide
```

> [Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a protocol standardizing interactions between AIs and systems.

Reviewer 3: ```suggestion
[Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a protocol that standardizes how AI systems interact with external systems.
```

> [Model Context Protocol (MCP)](https://modelcontextprotocol.io/docs/getting-started/intro) is a protocol standardizing interactions between AIs and systems.

Reviewer 3: ```suggestion

## What is MCP Servers

MCP ([Model Context Protocol](https://modelcontextprotocol.io/docs/getting-started/intro)] is a protocol that standardizes how AI systems interact with external systems.
```

While [AI actions](ai_actions_guide.md) integrate AI to the back office,

> [[= product_name =]]'s [MCP servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer an API usable by AI agents outside the system.

Reviewer 3: ```suggestion
While [AI actions](ai_actions_guide.md) integrate AI with the back office, [[= product_name =]]'s [MCP Servers](https://modelcontextprotocol.io/docs/learn/server-concepts) offer an API that can be used by AI agents from the outside of the system.
```

Also, I would insert a diagram, something that explains how our MCP servers "talk" to AI services?

<img width="1501" height="501" alt="Image" src="https://github.com/user-attachments/assets/1dafabb0-cc91-417a-a2af-367e30c4d4d5" />

Author: Done in 69b6fd8702578228e1c4f88233bf06d7548a9f99
See https://ez-systems-developer-documentation--3106.com.readthedocs.build/en/3106/ai/mcp/mcp_guide/#what-is-mcp-servers

Some AI agents could use directly REST API or GraphQL API if their users explain to them how to do it in prompts, in skill files, etc.
But MCP servers ease the discovery of the functionalities by AI agents and help them to interpret natural language prompts into actions on the system.
As MCP is a standard protocol, agents are already trained to use it.
With a singular REST API or GraphQL API, an agent can misunderstand the purpose of endpoints, hallucinate paths, and misshape parameters.

> With a standard MCP server, it can list the available tools and their parameters, learn how to use them, and pick the right one.

Reviewer 3: ```suggestion
Because MCP is a standard protocol, many agents are already trained to use it.
They can interact directly with REST or GraphQL APIs if their users provide detailed instructions through prompts, skill files, etc.
However, when facing a specific REST or GraphQL API, an agent may misunderstand the purpose of endpoints, hallucinate paths, or send incorrectly structured parameters.

MCP servers make the discovery of available capabilities much easier.
They help AI agents translate natural language prompts into concrete actions on the system.
An MCP server allows the agent to discover available tools, inspect their parameters, learn how to use them, and select the correct action.
```

> The MCP servers feature is an [LTS Update package](editions.md#lts-updates) available since v5.0.8 to all editions.

Reviewer 3: ```suggestion
## Availability

MCP Servers feature is an [LTS Update package](editions.md#lts-updates) available starting with the v5.0.8 in all Ibexa DXP editions.
```

> With the MCP servers feature, you can:

Reviewer 3: ```suggestion
## Capabilities

With the MCP Servers feature, you can:
```

- create MCP servers [by using YAML configuration](mcp_config.md#mcp-server-configuration)
- assign different tools, prompts, and resources to different MCP servers, varying them for each site and purpose
- use [built-in tools](mcp_config.md#built-in-tools) included in the package
- [create custom server capabilities](mcp_usage.md#create-capability-class) with PHP API

> MCP servers are configured per [repository](repository_configuration.md) and assigned to [SiteAccesses](siteaccess.md), allowing for flexible configurations adapted to different contexts.

Reviewer 3: ```suggestion
MCP servers are defined specifically for each [repository](repository_configuration.md) and assigned to individual [SiteAccesses](siteaccess.md) scopes.
This way you can build flexible configurations that match different contexts.
```
